source/extensions/filters/common/ratelimit/ratelimit_impl.cc

```diff
@@ -72,20 +72,28 @@ void GrpcClientImpl::onSuccess(
     span.setTag(Constants::get().TraceStatus, Constants::get().TraceOk);
   }
 
-  Http::HeaderMapPtr headers = std::make_unique<Http::HeaderMapImpl>();
-  if (response->headers_size()) {
+  Http::HeaderMapPtr response_headers_to_add = std::make_unique<Http::HeaderMapImpl>();
```

Comment:
@nandu-vinodan not your code, but here and below it would be better from a perf perspective to not allocate a header map unless we actually have headers to add. Do you mind doing a follow up? Thank you!

Reply:
Sure @mattklein123, made the changes: #8640